aot/README.md

```diff
@@ -1,3 +1,18 @@
 # Chicory AOT
 
 Experimental Wasm -> JVM bytecode AOT engine.
+
+To enable use the AotMachine factory when building the module:
+
+```java
+// ...
+import com.dylibso.chicory.aot.AotMachine;
+// ...
+
+var module = Module
+        .builder("mymodule.wasm")
+        .withMachineFactory(
+            instance -> new AotMachine(instance.module().wasmModule(), instance)
```

Comment:
Looking at this as a user, could we not just narrow AotMachine to only take the instance parameter?

Reply:
In the tests is slightly different:
https://github.com/dylibso/chicory/blob/530f6e5d0c03158613663f6c92c3631297fd4705/aot/src/test/java/com/dylibso/chicory/testing/TestModule.java#L79

That said, I think that this API is just instrumental to enable the usage of the test-gen  infrastructure.
I'm not against merging it, but we need a few more steps here to provide something "usable".
Cc. @danielperano @electrum

Reply:
Yeah, we should change the API to just take `Instance` for now. I put up a PR for that: https://github.com/dylibso/chicory/pull/416

The API really needs to be split so that we can compile separately from binding to the instance. Compiling only requires `com.dylibso.chicory.wasm.Module`. `Instance` is used afterwards to bind the instance to the compiled methods. This would allow us to have a Maven plugin that compiles the bytecode ahead of time to ship with the application, cache the compiled classes across JVMs, etc.